### PR TITLE
doc/sdk-cli-reference: Adding details for up/local command.

### LINKED ISSUE #266: Provide a detailed documentation of operator-sdk CLI
Currently, It might not be clear on what each `operator-sdk` CLI command does and when it should be used. Some of the commands are briefly explained in the [user_guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md) but not all.
We should have a detailed doc `sdk-cli-reference.md` to explain `operator-sdk` CLI usage.

### PR DESCRIPTION
**Pull request by shawn-hurley**
adds details and more examples for `operator-sdk up local` flags

Fixes #266

### CODE REVIEW

File: doc/sdk-cli-reference.md [hunk 1]
```diff
@@ -186,6 +186,20 @@ operator-sdk up local --kubeconfig "mycluster.kubecfg" \
...
 
+The below example will use the default kubeconfig, the default namespace environment var, and pass in flags for the operator.
+To use the operator flags, your operator must know how to handle the option. Below imagine an operator that understands the `resync-interval` flag.
+
+```bash
+operator-sdk up local --operator-flags "--resync-interval 10"
+```
+
+If you are planning on using a different namespace than the default, then you should use the namespace flag to change where the operator is watching for custom resources to be created.
+For this to work your operator must handle the `WATCH_NAMESPACE` environment variable. To do that you can use the [utility function](https://github.com/operator-framework/operator-sdk/blob/89bf021063d18b6769bdc551ed08fc37027939d5/pkg/util/k8sutil/k8sutil.go#L140) `k8sutil.GetWatchNamespace` in your operator.
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: Prefer to use `[utility function][utility_link]` and provide

`[utility_link]: https://github.com/operator-framework/operator-sdk/blob/89bf021063d18b6769bdc551ed08fc37027939d5/pkg/util/k8sutil/k8sutil.go#L140`
at end of the file.

File: doc/sdk-cli-reference.md [hunk 2]
```diff
@@ -186,6 +186,20 @@ operator-sdk up local --kubeconfig "mycluster.kubecfg" \
...
 ```
 
+The below example will use the default kubeconfig, the default namespace environment var, and pass in flags for the operator.
+To use the operator flags, your operator must know how to handle the option. Below imagine an operator that understands the `resync-interval` flag.
+
+```bash
+operator-sdk up local --operator-flags "--resync-interval 10"
+```
+
+If you are planning on using a different namespace than the default, then you should use the namespace flag to change where the operator is watching for custom resources to be created.
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: `should use the namespace flag than the default` to
`should use the --namespace flag than the default namespace`?

**Comment by fanminshi** [comment]
lgtm after nits.

File: doc/sdk-cli-reference.md (hunk 2 above)
Comment (shawn-hurley) [reply doc/sdk-cli-reference.md]: @fanminshi I'm sorry, but I am not following. 

> should use the namespace flag to change where the operator is watching for custom resources to be created.

Is this the part that you would like changed?

Comment (fanminshi) [reply doc/sdk-cli-reference.md]: sorry for the confustion, I meant 
`If you are planning on using a different namespace than the default, then you should use the namespace flag` ->
`If you are planning on using a different namespace than the default, then you should use the --namespace flag`

**Pull request merged**